docs/contributing.rst

```diff
@@ -1,7 +1,72 @@
 Contributing to sparse
 ======================
 
+General Guidelines
+------------------
 sparse is a community-driven project on GitHub. You can find our
 `repository on GitHub <https://github.com/mrocklin/sparse>`_. Feel
 free to open issues for new features or bugs, or open a pull request
 to fix a bug or add a new feature.
+
+If you haven't contributed to open-source before, we recommend you read
+`this excellent guide by GitHub on how to contribute to open source
+<https://opensource.guide/how-to-contribute/>`_. The guide is long,
+so you can gloss over things you're familiar with.
+
+Guidelines Specific to this Project
+-----------------------------------
+If you're not already familiar with it, we follow the `fork and pull model
+<https://help.github.com/articles/about-collaborative-development-models/>`_
+on GitHub.
+
+To get started, you can do something along the lines of the following, after
+creating a fork:
+
+.. code-block:: bash
+   git pull https://github.com/<username>/sparse.git
+   git remote add upstream https://github.com/mrocklin/sparse.git
+   git install -e .[docs, tests]
+
+Then you can make changes to a branch on your fork and open a pull request
+on GitHub. There, it can be discussed, and if the community decides it's
+best, it will be merged into the main codebase. Afterwards, you can
+`sync your fork on GitHub <https://help.github.com/articles/syncing-a-fork/>`_
+
+Running/Adding Unit Tests
+-------------------------
+It is best if all new functionality and/or bug fixes have unit tests added
+with each use-case.
+
+Since we support both Python 2.7 and Python 3.5 and newer, it is recommended
+to test with at least these two versions before committing your code or opening
+a pull request. We use `pytest <https://docs.pytest.org/en/latest/>`_ as our unit
+testing framework, with the pytest-cov extension to check code coverage and
+pytest-flake8 to check code style. You don't need to configure these extensions
+yourself. Once you've configured your environment, you can just :code:`cd` to
+the root of your repository and run
+
+.. code-block:: bash
+   py.test
```

Comment:
I usually have to run `py.test sparse`

Also, do you need a blank line between code-block and py.test?  I've always put this there, but I'm not sure if its necessary.

Reply:
Yes, I put those in in a later commit. It's necessary.

`py.test` works for me at least. I just tested... Are you running it from the root of the repo?